### PR TITLE
add AWS CLI, update SBT, more reliably download PGP keys, …

### PR DESCRIPTION
**Pull request by bpholt**
…and pre-download several versions of Scala

### CODE REVIEW

File: fake-project/project/build.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+sbt.version=0.13.16
```
Comment (lochnesh) [inline fake-project/project/build.properties:1]: Should this be `1.0.4`?

File: Dockerfile [hunk 2]
```diff
@@ -4,50 +4,58 @@ LABEL org.label-schema.vcs-url="https://github.com/Dwolla/jenkins-agent-docker-n
...
 
-RUN printf "set scalaVersion := \"${SCALA_VERSION}\"\nupdate-sbt-classifiers\nsbtVersion\n" | sbt && \
-    rm -rf $JENKINS_HOME/project/ $JENKINS_HOME/target/
+RUN curl -sL https://raw.githubusercontent.com/creationix/nvm/${NVM_VERSION}/install.sh | /bin/bash
 
 RUN git config --global user.email "dev+jenkins@dwolla.com" && \
     git config --global user.name "Jenkins Build Agent"
 
+RUN cd $JENKINS_HOME/fake-project && \
+    echo sbt.version=0.13.16 > project/build.properties && \
```
Comment (lochnesh) [inline Dockerfile:55]: Should this be `1.0.4`?  Should it use the ENV variable `SBT_VERSION`?

**Review by lochnesh — CHANGES_REQUESTED** [review]

File: fake-project/project/build.properties (hunk 1 above)
Comment (bpholt) [reply fake-project/project/build.properties:1]: This file doesn't matter; it's overwritten by the script in `Dockerfile`. It's only here to make it work reasonably when testing locally.

File: Dockerfile (hunk 2 above)
Comment (bpholt) [reply Dockerfile:55]: No, it's intentionally pulling both 1.0.4 and 0.13.16 so both are available to jobs without having to be downloaded at runtime.

Comment (lochnesh) [reply Dockerfile:55]: Thanks for the reminder.  I think I knew this at one point, but I forgot.

**Review by lochnesh — APPROVED** [review]

**Pull request merged**